lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -1185,6 +1188,95 @@ LinearLayout SliceEncodingAttr::toLinearLayout(ArrayRef<int64_t> shape) const {
                       llvm::to_vector(sliceLL.getOutDimNames()));
 }
 
+LinearLayout tensorMemoryToLinearLayout(ArrayRef<int64_t> shape,
+                                        TensorMemoryEncodingAttr encoding) {
+  // We model tensorMemory as a 2D layout. Interestingly enough, we craft the
+  // map as going from the matrix into the tensorMemory, rather than the other
+  // way around. We do this because of the following reasons:
+  // - When packed=True, the coordinate (0, 0) would have to map to BOTH
+  // elements (0, 0) and (0, 1)
+  // - For blockM=64, when we have a tensor of shape 64xblockN, we leave every
+  // other 16 rows empty.
+  //   This cannot be modelled as a map from TMEM into the tensor as you don't
```

Comment:
Oh this is interesting

Reply:
Just learned that ldmatrix now supports loading packed f4 in smem into padded registers, similarly to how fp4 TMA loads packed fp4 in gmem into padded smem: 

https://docs.nvidia.com/cuda/parallel-thread-execution/#warp-level-matrix-instructions-ldmatrix
<img width="1128" height="276" alt="Screenshot 2025-08-04 at 8 58 18" src="https://github.com/user-attachments/assets/2bf7e2d2-3ab5-4d8c-b584-b8e8d0aa7935" />

This is used for sm120 mxfp dot with `mma.sync`.

Reply:
Interesting. We'll support the `b8` variants once the become available in nvvm (see https://github.com/triton-lang/triton/pull/7542 and https://github.com/llvm/llvm-project/pull/148783).
I could also throw in `b4x16_p64` in case we ever want to load `f4`. That being said, I don't know of any use case ATM.